packages/salesforcedx-vscode-core/src/conflict/persistentStorageService.ts

```diff
@@ -35,22 +35,26 @@ export class PersistentStorageService {
     return PersistentStorageService.instance;
   }
 
-  public getPropertiesForFile(fileName: string): ConflictFileProperties | undefined {
-    return this.storage.get<ConflictFileProperties>(fileName);
+  public getPropertiesForFile(key: string): ConflictFileProperties | undefined {
+    return this.storage.get<ConflictFileProperties>(key);
   }
 
-  public setPropertiesForFile(fileName: string, value: ConflictFileProperties | undefined) {
-    this.storage.update(fileName, value);
+  public setPropertiesForFile(key: string, conflictFileProperties: ConflictFileProperties | undefined) {
+    this.storage.update(key, conflictFileProperties);
   }
 
-  public setPropertiesForFiles(fileProperties: FileProperties | FileProperties[]) {
+  public setPropertiesForFilesRetrieve(fileProperties: FileProperties | FileProperties[]) {
```

Comment:
Renamed the function to `setPropertiesForFilesRetrieve` because the deploy side would follow a different strategy which fetches `type` and `fullName` from `sourceComponent` and `lastModifiedDate` from `MetadataApiDeployStatus`. I’ll create a separate PR for deploy to avoid confusion.

Reply:
I don't mind this idea! I think there is a challenge we will want to discuss which is that we would like to use `SourceComponent.type.id` as the first half of our keys. However, `FileProperties` do not come naturally correlated with `SourceComponent`s. Not sure what the best solution is.

Reply:
Update: after a good amount of thought we have decided to do this just as Violet originally had it, using the type name and not the id as the key for the cache. I don't want to do an official "approving review" but I am now on board with the changes in this PR!

Reply:
Discussed offline and would also push `setPropertiesForFilesDeploy` in this PR.